shared/reports/carryforward.py

```diff
@@ -66,19 +65,16 @@ def generate_carryforward_report(
     Returns:
         EditableReport: A new report with only the info related to `flags` on it, as described above
```

Comment:
[very nit] now it returns `Report`, although they might be the same thing?

Docs also mention that it "creates a new report", but the code suggests all changes are done in-place, and the same report is returned.

Reply:
I wonder if that could pose a problem... say we use `commit.parent` report to build a "new" carryforward report for `commit`. 

Would we save the modified `commit.parent` report back with the changes? Probably not, but if it's possible it might be a good idea to explicit that in the documentation

Reply:
Thanks for reminding me about this. Indeed this is a behavior change, and I need to update the docs, and make sure that the calling code is aware of this.